crates/block-producer/src/rpc_client.rs

```diff
@@ -155,7 +155,13 @@ impl RPCClient {
                     )
                     .await?,
             )?;
+
+            if cells.last_cursor.is_empty() {
+                println!("no payment cell");
+                return Ok(collected_cells);
```

Comment:
We should return an error since we haven't collected enough capacity.

Reply:
Ok